staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -905,7 +905,24 @@ func (c *Cacher) dispatchEvents() {
 	bookmarkTimer := c.clock.NewTimer(wait.Jitter(time.Second, 0.25))
 	defer bookmarkTimer.Stop()
 
+	// The internal informer populates the RV as soon as it conducts
+	// The first successful sync with the underlying store.
+	// The cache must wait until this first sync is completed to be deemed ready.
+	// Since we cannot send a bookmark when the lastProcessedResourceVersion is 0,
+	// we poll aggressively for the first RV before entering the dispatch loop.
+	if err := c.ready.wait(wait.ContextForChannel(c.stopCh)); err != nil {
```

Comment:
Why do we need it? I suggest removing it.

Reply:
We have an extremely tight loop below. 

Before hammering the CPU, we should check if the cacher has been synchronised so that we can read the current value of the resource version.

Reply:
What is your concern here? Does calling a well-known function really make this code more complicated ?:)
There is no point in calling `getResourceVersion` before the cacher synchronies.